control/Director.php

```diff
@@ -384,7 +384,7 @@ protected static function handleRequest(SS_HTTPRequest $request, Session $sessio
 					try {
 						$result = $controllerObj->handleRequest($request, $model);
 					} catch(SS_HTTPResponse_Exception $responseException) {
-						HTTPCacheControl::singleton()->disableCache(true);
+						HTTPCacheControl::singleton()->disableCache();
```

Comment:
If you are using `enableCache($force=true)`, which will be common for websites which want caching despite some pages starting a session for CSRF tokens, this means 5xx errors will be cached. Which wasn't the intention of the dev, and makes using this mode dangerous - it can elevate a one off server error to cached response for all visitors for minutes or hours.

Reply:
Hm true. The other issue is what if we want enableCache(true) for errors explicitly, which is why I removed the force here.

Reply:
How about we document not using force = true in usercode?

Reply:
Here are the docs on what requests are cachable, which includes some https://developer.mozilla.org/en-US/docs/Glossary/cacheable

Some error codes are cacheable by default (404, 405, 410, 414, and 501) - I don't think it's sensible to say that if a dev used `enableCache($force = true)` they didn't really intend the cache to be enabled... they shouldn't be enabling the cache, then ;)

Reply:
There's been a bunch of discussion on Slack, the short of it is that `enableCache($force=true)` in user code will be quite common due to websites with CSRF tokens, but otherwise cacheable content (on pages without them). So we need to ensure there's a safe way to use it, and document it. Just saying "well, they shouldn't have used this setting" goes against the spirit of this work - we're providing safe defaults for common use cases, without requiring an expert level understanding of either caching headers or core behaviour influencing cacheability outside of your own code.

@dhensby Thanks for highlighting the HTTP spec around status codes. I've tracked it down in the actual spec wording (https://tools.ietf.org/html/rfc7231#section-6.6.2), it doesn't say that 500 isn't cacheable, but it *does* mention it for 501 - so non-cacheable 500s are implied. I'm not sure if it declares a priority between sending explicit `Cache-Control` caching headers on 500s, and the general spec rule of "500s aren't cacheable". So I still don't feel allowing devs to send cache headers for `isError()` (by using `disableCache()` rather than `disableCache(true)`).